java/org/apache/catalina/filters/CsrfPreventionFilter.java

```diff
@@ -53,6 +58,25 @@ public class CsrfPreventionFilter extends CsrfPreventionFilterBase {
 
     private String nonceRequestParameterName = Constants.CSRF_NONCE_REQUEST_PARAM;
 
+    private boolean enforce = true;
+
+    private Collection<Predicate<String>> noNoncePatterns = DEFAULT_NO_NONCE_URL_PATTERNS;
+
+    private static final Collection<Predicate<String>> DEFAULT_NO_NONCE_URL_PATTERNS;
+
+    static {
+        ArrayList<Predicate<String>> defaultNoNonceURLPatterns = new ArrayList<>();
+
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".css"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".js"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".gif"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".png"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".jpg"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".svg"));
+
+        DEFAULT_NO_NONCE_URL_PATTERNS = Collections.unmodifiableList(defaultNoNonceURLPatterns);
```

Comment:
What about `.ico`?

Reply:
Sure. I'm happy to add some obvious things. I thought `.ico` was dead, actually.

Reply:
Might this be better handled by looking at content type rather than file extension?

Reply:
Is this really dead https://en.wikipedia.org/wiki/Favicon#How_to_use?

Reply:
I just assumed everyone used file formats other than `.ico` for their "Favicon". It's not "favicon" that's dead... it's `.ico`. But I can add it to the default. Remember: these are just the _defaults_. Anyone can change them vi configuration.

Reply:
> Might this be better handled by looking at content type rather than file extension?

All we have to go on is the path of the file, so I don't think we can check the content-type. Remember, this is primarily being used inside of `CsrfResponseWrapper.encodeURL(String)` where we know nothing about the file other than the path.

Reply:
I've added `.ico` but I want to ensure I understand @markt-asf's comment about the content-type before I "resolve" this review.

Reply:
Use `ServletContext.getMimeType(String)`? My thinking was a filter for `image/*` was a lot more compact and somewhat future proof. You could also exclude `*font*` and so on.

Reply:
Is is possible that an image is created on the fly containing some kind of secret (e.g., QR code, OTP) which should be protected?

Reply:
@markt-asf Sure, this could work. I could use a new type of matcher such as `mime:[whatever]` and then use the match based upon MIME type instead of filename. Then you can mix-and-match.

Reply:
@michael-o All of this stuff can be configured-around by the user. We are just talking about the defaults, here. If you want to serve static content from `/static/*.jpg` but dynamic files from `/dynamic/*.jpg` which need protection, then you can set up a regular-expression-based check.

An out-of-the-box default should work for "a great many environments" not "every single conceivable environment".

Reply:
I think this has been resolved in ef54a1e44fe32aa0ec0fd0559726b214275046aa and d0433b10c98b90cdca573a14c4eb64dff8bdc980.

Reply:
Agree